ImageCollectionViewController.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  ImageCollectionViewController.swift
+//  FlickrViewer
+//
+//  Created by Kirill Shteffen on 04/07/2018.
+//  Copyright © 2018 BlackBricks. All rights reserved.
+//
+
+import UIKit
+
+
+class ImageCollectionViewController: UICollectionViewController {
+    
+    typealias FlickrResponse = (NSError?, [Commentary]?) -> Void
+    
+    var photos: [FlickrPhoto]=[]
+    var commentaries:[Commentary]=[]
+    
+    // MARK: UICollectionViewDataSource
+    
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return photos.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "ImageCollectionViewCell", for: indexPath) as? ImageCollectionViewCell
+        cell!.setupWithPhoto(flickrPhoto: photos[indexPath.row])//}
```

Comment:
we can avoid using force unwrap 
https://swiftbook.ru/content/control-flow/early-exit/

Reply:
guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "ImageCollectionViewCell", for: indexPath) as? ImageCollectionViewCell else {return UICollectionViewCell.init()}
is this a suitable line of code?

Reply:
looks good enough. And I think you can use UICollectionViewCell() instead of UICollectionViewCell.init()